bgcval2/functions/ice.py

```diff
@@ -0,0 +1,228 @@
+
+
+def calcTotalIceAreaN(nc,keys, **kwargs): # North
+    if 'areafile' not in kwargs.keys():
+        raise AssertionError("calcTotalIceAreaN:\t Needs an `areafile` kwarg to run calculation.")
+    try:
+        maskname = kwargs['maskname']
+    except:
+        maskname = 'tmask'
+
+    if not loadedArea:
+        loadDataMask(kwargs['areafile'], maskname)
+
+    arr = nc.variables[keys[0]][:].squeeze() * area
+    return np.ma.masked_where((tmask==0) + (lat<0.), arr).sum() / 1E12
+
+
+def calcTotalIceAreaS(nc, keys, **kwargs): # South
+    if 'areafile' not in kwargs.keys():
+        raise AssertionError("calcTotalIceAreaS:\t Needs an `areafile` kwarg to run calculation.")
+    try:
+        maskname = kwargs['maskname']
+    except:
+        maskname = 'tmask'
+    if not loadedArea:
+        loadDataMask(kwargs['areafile'], maskname)
+
+    arr = nc.variables[keys[0]][:].squeeze() * area
+    return np.ma.masked_where((tmask==0)+(lat>0.), arr).sum() / 1E12
+
+
+def calcTotalIceExtent(nc,keys, **kwargs):    #Global
+    if 'areafile' not in kwargs.keys():
+        raise AssertionError("calcTotalIceExtent:\t Needs an `areafile` kwarg to run calculation.")
+    try:
+        maskname = kwargs['maskname']
+    except:
+        maskname = 'tmask'
+    if not loadedArea:
+        loadDataMask(kwargs['areafile'], maskname)
+
+    try:
+        minIce = float(kwargs['minIce'])
+    except:
+        minIce = 0.15
+
+    if not loadedArea:
+        loadDataMask(kwargs['areafile'],maskname)
+    
+    return np.ma.masked_where((tmask==0) + (nc.variables[keys[0]][:].squeeze()<minIce), area).sum() / 1E12
+
+
+def calcTotalIceExtentN(nc, keys, **kwargs): # North
+    if 'areafile' not in kwargs.keys():
+        raise AssertionError("calcTotalIceExtentN:\t Needs an `areafile` kwarg to run calculation.")
+    try:
+        maskname = kwargs['maskname']
+    except:
+        maskname = 'tmask'
+    if not loadedArea:
+        loadDataMask(kwargs['areafile'], maskname)
+
+    try:
+        minIce = float(kwargs['minIce'])
+    except:
+        minIce = 0.15
+
+    return np.ma.masked_where((tmask==0) + (nc.variables[keys[0]][:].squeeze()<minIce) + (lat<0.), area).sum() / 1E12
+
+
+def calcTotalIceExtentS(nc, keys, **kwargs): # South
+
+    if 'areafile' not in kwargs.keys():
+        raise AssertionError("calcTotalIceExtentS:\t Needs an `areafile` kwarg to run calculation.")
+    try:
+        maskname = kwargs['maskname']
+    except:
+        maskname = 'tmask'
+    if not loadedArea:
+        loadDataMask(kwargs['areafile'], maskname)
+
+    try:
+        minIce = float(kwargs['minIce'])
+    except:
+        minIce = 0.15
+
+    return np.ma.masked_where((tmask==0) + (nc.variables[keys[0]][:].squeeze()<minIce) + (lat>0.), area).sum()/1E12
+
+
+icedetails = {}
+def loadArea(gridfn):
+    global icedetails
+    nc = dataset(gridfn, 'r')
+    lat    = nc.variables['lat'][:]
+    try:
+        area   = nc.variables['area' ][:]
+    except:
```

Comment:
OK - this is really under the hood - try avoiding these generic excepts (I made of a reference to this type of handling above too) - maybe not change stuff here, since it works - but for the future - handle exceptions better :+1: